docs/protocol.md

```diff
@@ -1,211 +1,370 @@
+
+#### 3.2. Setup and periodic replenishment of $n$ ephemeral keys
+
+Each journalist $J$ MUST generate and maintain a pool of $n$ ephemeral keys.
+For each key:
+
+| Journalist                                                                              |                                                                    | Server       |
+| --------------------------------------------------------------------------------------- | ------------------------------------------------------------------ | ------------ |
+| $`(J_{edh,sk}, J_{edh,pk}) \gets^{\$} \text{Gen}()`$                                    |                                                                    |              |
+| $`(J_{ekem,sk}, J_{ekem,pk}) \gets^{\$} \text{Gen}()`$                                  |                                                                    |              |
+| $`(J_{epke,sk}, J_{epke,pk}) \gets^{\$} \text{Gen}()`$                                  |                                                                    |              |
+| $`\sigma^J \gets^{\$} \text{Sign}(J_{sig,sk}, (J_{edh,pk}, J_{ekem,pk}, J_{epke,pk}))`$ |                                                                    |              |
+|                                                                                         | $`\longrightarrow J_{edh,pk}, J_{ekem,pk}, J_{epke,pk}, \sigma^J`$ | Save for $J$ |
+
+### 4. Source
+
+To begin each session, a source MUST enter (on their first visit) or reenter (on
+a subsequent visit) some $passphrase$:
+
+| Source                                                                                          |
+| ----------------------------------------------------------------------------------------------- |
+| $`S_{dh,sk} \Vert S_{fetch,sk} \Vert S_{pke,sk} \Vert S_{kem,sk} \gets \text{KDF}(passphrase)`$ |
+
+## Messaging protocol
+
+SecureDrop is a first-contact protocol between an unknwn party (an anonymous
+source) and well-known parties (journalists). Only a source can initiate a
+conversation.
+
+### 5. Source fetches keys and verifies their authenticity
+
+For some newsroom $NR$ and all its enrolled journalists $J^i$:
+
+| Source                                                    |                                                                             | Server                                               |
+| --------------------------------------------------------- | --------------------------------------------------------------------------- | ---------------------------------------------------- |
+|                                                           | $\longrightarrow$ request keys for $NR$                                     |                                                      |
+|                                                           | $`NR_{sig,pk}, \sigma^{FPF} \longleftarrow`$                                |                                                      |
```

Comment:
at this stage, could we verify the sig and terminate early if it fails? i.e. we could move up $\text{Vfy}(FPF_{sig,pk}, NR_{sig,pk}, \sigma^{FPF})$?